### ISSUE TITLE
🐞incorrect validation error when using client env variable as default

### ISSUE THREAD

**Issue by shykes** [dagger/dagger#2363]
## Problem

When an action uses a client's environment variable as its *default*, then `dagger do` will incorrectly fail at the validation stage ("xyz not set").

NOTE:

* If the action references the client's env variable directly (not as a default value), then the problem does not occur
* If the action uses a default value other than the client env variable (ie. a string literal, or reference to a concrete string), then the problem does not occur

### What I did

1. Setup repro environment

```
cd $(mktemp -d)
dagger project init
```

2. Write repro configuration

```
cat > repro.cue <<-EOF
package main

import (
        "dagger.io/dagger"
        "dagger.io/dagger/core"
)

dagger.#Plan & {
        client: env: USER: string
        actions: {
                deploy: {
                        site: string | *client.env.USER
                        command: core.#Nop & {
                                input: dagger.#Scratch
                        }
                }
        }
}
EOF
```

3. Install dependencies

```
dagger project update
```

4. Run deploy action

```
dagger do deploy
```

### What I expected to see

Output of step 4:

```
$ dagger do deploy
[✔] actions.deploy.command                                                                                                                        0.0s
```

### What I see instead

```
$ dagger do deploy
failed to load plan: "actions.deploy.site" is not set:
    ./repro.cue:11:3
    ./repro.cue:12:4


Usage: 
  dagger do [flags]


Flags:
      --cache-from stringArray   External cache sources (eg. user/app:cache, type=local,src=path/to/dir)
      --cache-to stringArray     Cache export destinations (eg. user/app:cache, type=local,dest=path/to/dir)
  -h, --help                     help for do
      --no-cache                 Disable caching
  -p, --plan string              Path to plan (defaults to current directory) (default ".")
      --platform string          Set target build platform (requires experimental)
  -w, --with stringArray
Global Flags:
      --experimental        Enable experimental features
      --log-format string   Log format (auto, plain, tty, json) (default "auto")
  -l, --log-level string    Log level (default "info")
```

**Comment by aluzzardi** [comment]
Ok this is pretty bad.

According to CUE:
- `actions.deploy.site` doesn't have a default value (`Value.Default()` returns `false`)
- `string | *client.env.USER` is not an expression (`Value.Expr()` returns `cue.NoOp`)

@jlongtine Help appreciated if you can spare the time :)

**Comment by aluzzardi** [comment]
Fixed in #2366
